src/components/can_cooperation/src/command/get_interior_vehicle_data_request.cc

```diff
@@ -125,6 +98,40 @@ void GetInteriorVehicleDataRequest::OnEvent(
   }
 }
 
+
+void GetInteriorVehicleDataRequest::ProccessSubscription() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  Json::Value params;
+
+  Json::Reader reader;
+  reader.parse(message_->json_message(), params);
+
+  if (IsMember(params, kSubscribe)){
+    bool subscribe = params[kSubscribe].asBool();
+
+    if (!response_params_.isMember(kIsSubscribed)) {
+      if (subscribe){
+        response_params_[kIsSubscribed] = false;
+      } else {
+        response_params_[kIsSubscribed] = true;
+      }
+    } else {
+      CANAppExtensionPtr extension = GetAppExtension(app());
+
+      bool isSubscribed = params[kIsSubscribed].asBool();
```

Comment:
subscribe and isSubscribed are very similar. I suggest to use more different names. isNowSubscribed and needSubscribe maybe. You know.